docs/book/src/reference/attributes.md

```diff
@@ -0,0 +1,52 @@
+# Attributes
+
+The Sway compiler supports a list of attributes that perform various operations that are useful for building, testing and documenting Sway programs. Below is a list of all available attributes:
+
+## Allow
+
+The `#[allow(dead_code)]` attribute overrides the check for dead code so that violations will go unreported.
+
+## Doc
+
+The `#[doc(..)]` attribute specifies documentation.
+
+Line doc comments beginning with exactly three slashes `///`, are interpreted as a special syntax for doc attributes. That is, they are equivalent to writing `#[doc("...")]` around the body of the comment, i.e., `/// Foo` turns into `#[doc("Foo")]`
+
+Line comments beginning with `//!` are doc comments that apply to the module of the source file they are in. That is, they are equivalent to writing `#![doc("...")]` around the body of the comment. `//!` module level doc comments should be at the top of Sway files.
+
+Documentation can be generated from doc attributes using `forc doc`.
```

Comment:
This may be a good place to have linked to the documentation about `forc doc` since this command will only generate the documentation and not open it.